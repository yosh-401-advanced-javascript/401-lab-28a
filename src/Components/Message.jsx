import React from 'react';


class Message extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      text: 'Hello World',
      name: this.props.name,
      number: this.props.number,
    };
  }

  render() {
    return (
        <div>
          {this.state.text}
          <div>{this.props.name}</div>
          <p>{this.props.number}</p>
        </div>
    );
  }
}


export default Message;
