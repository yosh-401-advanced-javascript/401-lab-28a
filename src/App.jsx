
import React from 'react';
import Message from './Components/Message'


class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      name: 'Han Solo',
      number: '4'
    };

  }


  render() {
    return (
        <div>
          <Message
              name={this.state.name}
              number={this.state.number}
          />
        </div>
    )
  }
}

export default App
