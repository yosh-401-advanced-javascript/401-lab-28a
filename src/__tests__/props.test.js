import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App'
import Message from '../Components/Message.jsx'

// describe('Testing Counter', () => {
//   it('is alive at application start', function() {
//     let app = shallow(<App />);
//     expect(app.find('div').exists()).toBeTruthy();
//   });
// });
//
// it('should render correctly', () => {
//   const tree = renderer.create(<App />).toJSON();
//   expect(tree).toMatchSnapshot();
// });


describe('Testing App ', () => {
  it('should should render correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe('Testing Message ', () => {
    it('should should render correctly', () => {
      const tree = renderer.create(<Message />).toJSON();
      expect(tree).toMatchSnapshot();

    });
  });

});

