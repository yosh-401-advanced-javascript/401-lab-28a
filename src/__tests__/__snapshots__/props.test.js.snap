// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App  Testing Message  should should render correctly 1`] = `
<div>
  Hello World
  <div />
  <p />
</div>
`;

exports[`Testing App  should should render correctly 1`] = `
<div>
  <div>
    Hello World
    <div>
      Han Solo
    </div>
    <p>
      4
    </p>
  </div>
</div>
`;
